jte-spring-boot-starter-2/src/main/java/gg/jte/springframework/boot/autoconfigure/JteAutoConfiguration.java

```diff
@@ -39,8 +38,8 @@ public JteViewResolver jteViewResolver(TemplateEngine templateEngine) {
     @ConditionalOnMissingBean(TemplateEngine.class)
     public TemplateEngine jteTemplateEngine() {
 
-        if (jteProperties.isProductionEnabled(environment)) {
-            // Templates will be compiled by the maven build task
+        if (jteProperties.usePreCompiledTemplates()) {
```

Comment:
Is it still possible to have `usePrecompiledTemplates = true` in production and `usePrecompiledTemplates = false` during local development? Because this setting is going to be what 99% of all user will want to have.

From a security point of view, it would be bad to encourage users to run with hot-reloadable templates in production.

Reply:
Why is it bad?
The best way would probably to do the template resolving at startup in production.

Reply:
It is bad because it requires a JDK at runtime, not just a JRE. From a security point of view it is better to precompile templates on the build server, instead of the production server. The same way Kotlin/Java sources are compiled on the build server and not on prod.

> Just Files.walk trough the directory and resolve them all on startup?

What do you mean by that?

Reply:
Hey casid, 
sorry for the late reply. I think I have different solution that is a bit more work but would remove the necessity of using a build plugin.

I'm currently building an annotation processor for preprocessing of my html templates for Spring ViewComponent. This is primarily for HTMX attributes but I need to compile the JTE Templates aswell.

https://github.com/tschuehly/spring-view-component/blob/v0.6.2-SNAPSHOT/core/src/main/kotlin/de/tschuehly/spring/viewcomponent/core/processor/ViewComponentProcessor.kt

For the JTE Spring Boot Starter I think I could create an annotation processor that targets the @SpringBootApplication Annotation and then compiles the whole template folder at build time into the build/classes for gradle and target/classes for maven

For Hot Reload at development time I would implement a FileChangeListener that recompiles the changed file into the build folder.
https://github.com/tschuehly/spring-view-component/blob/v0.6.2-SNAPSHOT/core/src/main/kotlin/de/tschuehly/spring/viewcomponent/core/processor/ViewComponentChangeListener.kt

I did a bit of digging into the JTE sources and I think I would just create a precompiled TemplateEngine and then use the gg.jte.compiler.TemplateCompiler.precompile() to compile a single template right?

What are your thoughts on that solution @casid?

Reply:
@casid 
I was look at the jstachio docs and they also use the annotation processor:

> JStachio does its type checking and code generation with the Java annotation processor which is built into the JDK. JTE and Rocker require pre-compiling with a build plugin. Thus you cannot precompile easily if your build system is not supported. JTE, Rocker, and JSP do not actually read the symbolic tree of the Java models but rather just translate the code to Java.

https://github.com/jstachio/jstachio/blob/main/why_jstachio_is_better.md

Reply:
I don't understand what the annotation processor should accomplish, as jte doesn't make use of annotations like jstachio does. Wouldn't this just introduce another layer of magic?

Both hot reload and precompiling templates work with the current spring boot starter already, no?

Maybe what you want is more specific to the Spring View Component library rather than to a general purpose spring boot starter plugin?

Reply:
Well, the annotation processor would solve this problem:
> Is it still possible to have `usePrecompiledTemplates = true` in production and `usePrecompiledTemplates = false` during local development? Because this setting is going to be what 99% of all user will want to have.
> 
> From a security point of view, it would be bad to encourage users to run with hot-reloadable templates in production.

You would need to have just one annotation in your whole project to enable it. With Spring you could just use @SpringBootApplication

The Annotation processor could replace the gradle/maven plugin. 

If you are not open to that idea I would just update the docs, with a reference that you need to use the maven/gradle plugin in production.

Reply:
This problem is already solved in the current version of the Spring Boot Starter. Maybe we should just keep it this way?

```
if (jteProperties.isProductionEnabled(environment)) {
    // Templates will be compiled by the maven build task
```

Updating the documentation with a hint to maven/gradle would be very good.

Reply:
IMO this hides the behaviour and makes it less obvious, that's why I changed it. 

It is not guaranteed that in production the profile is "prod", that's why I think with an update to the Readme with:
> In Production you should use the Gradle/Maven Plugin and set usePreCompiledTemplates to true, as this doesn't require a JDK at runtime and is safer

Reply:
I've adjusted the README and the autoconfiguration. Users will need to explicitly specify development mode or precompiled templates. 
If they do not the application will not start

Reply:
Nice!